go-controller/pkg/cni/cniserver.go

```diff
@@ -151,4 +173,6 @@ func (s *Server) handleCNIRequest(w http.ResponseWriter, r *http.Request) {
 			logrus.Warningf("Error writing %s HTTP response: %v", req.Command, err)
 		}
 	}
+
+	metricCNIRequestDuration.WithLabelValues(string(req.Command), hasErr).Observe(time.Since(startTime).Seconds())
 }
```

Comment:
@squeed this will not track the correct duration for the case where the interfaces are created and configured in the CNI Shim instead of the CNI Server.

Reply:
For sure - this is just the cniserver request duration. We can't export metrics from the CNI shim itself, since it's not a long-lived process.

However, the CNI shim doesn't do much; just some json shuffling. I'm not worried about it's added latency.

Reply:
@squeed we have an option to create and configure interfaces for pods in the CNI shim itself. In this mode, CNI Server acts as a proxy to fetch Pod annotation from K8s API server. This mode enables us to run ovnkube-node Pod with least privileges (doesn't require `privilege: true` to be set). You can enable this mode by using `--unprivileged-mode` or some such.

I think we can integrate this PR for now and see how we can provide the same metrics for the aforementioned mode as well.